obj_gen.cpp

```diff
@@ -54,26 +54,40 @@ void random_generator::set_seed(int seed)
 #endif
 }
 
-unsigned int random_generator::get_random()
+unsigned long long random_generator::get_random()
 {
     int rn;
+    unsigned long long llrn;
 #ifdef HAVE_RANDOM_R
     int ret = random_r(&m_data_blob, &rn);//max is RAND_MAX
     assert(ret == 0);
+
+    llrn = rn;
+    llrn = llrn << 32; // move to upper 32bits
+
+    ret = random_r(&m_data_blob, &rn);//max is RAND_MAX
+    assert(ret == 0);
+    llrn |= rn; // set lower 32bits
 #elif (defined HAVE_DRAND48)
     rn = nrand48(m_data_blob); // max is 1<<31
+    llrn = rn;
+    llrn = llrn << 32; // move to upper 32bits
+
+    rn = nrand48(m_data_blob); // max is 1<<31
+    llrn |= rn; // max is 1<<31
 #else
     #error no random function
 #endif
-    return rn;
+    return llrn;
 }
 
-unsigned int random_generator::get_random_max() const
+unsigned long long random_generator::get_random_max() const
 {
 #ifdef HAVE_RANDOM_R
-    return RAND_MAX;
+    unsigned long long rand_max = RAND_MAX;
+    return (rand_max << 32) | RAND_MAX;
```

Comment:
Not really important,  but you can do ((uint64_t)RAND_MAX << 32) | RAND_MAX; instead.